corehq/apps/saved_reports/models.py

```diff
@@ -744,6 +746,10 @@ def get_secret(self, email):
         return hashlib.sha1((uuid + email).encode('utf-8')).hexdigest()[:20]
 
     def send(self):
+        if not self.owner:
+            self.delete()
```

Comment:
Is this a hard deletion? If yes, is there an audit trail that allows someone to figure out how/why/when the report was deleted?

I think it's supposed to be impossible to reach this code in a production environment, but it seems like it would be really bad if a domain was depending on a report, and it somehow got automatically deleted (and simply disappeared) because the original owner was deleted.

Reply:
Yeah good question. I viewed this as adding onto [existing behavior](https://github.com/dimagi/commcare-hq/blob/a41c26e0567c082644c274b3c2118921df9471cc/corehq/apps/saved_reports/models.py#L753-L761) (to your point, adding a severe edge case that I don't anticipate). We have a ticket in the backlog to handle this better as it has resulted in confusion in the past: https://dimagi.atlassian.net/browse/SAAS-15918